aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/WithCrt.kt

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.sdk.kotlin.crt
+
+/**
+ * A mixin class used to ensure CRT is initialized before the class is invoked
+ */
+public open class WithCrt {
+    init {
+        CRT.initRuntime { }
+    }
+}
```

Comment:
**Question**: I'm concerned with this approach for two reasons:
1. Kotlin doesn't support true mixin classes so this pattern prevents inheritors from using any other superclass. Given that this functionality isn't _truly_ polymorphic it seems prudent to leave the option open.
1. This initializes the CRT runtime at object instantiation rather than when CRT functionality is actually necessary. If some code just instantiates a new `Crc32` but never uses any methods on it, we'll've wasted cycles unnecessarily.

Could we move the initialization closer to the site of use?

Reply:
> Could we move the initialization closer to the site of use?

Sure, we'll just have to take care to initialize CRT everywhere we use it, which we currently don't do.

Reply:
> If some code just instantiates a new Crc32 but never uses any methods on it, we'll've wasted cycles unnecessarily.

This is true but in practice CRT is only initialized once, subsequent calls to `CRT.initRuntime { }` are no-ops. I'm assuming users of aws-sdk-kotlin Kotlin/Native will need to initialize CRT at some point

Reply:
We spoke offline and agreed to continue this approach for now. Ideally we would create a singleton object `CrtLib` and use cinterop to generate extensions for it (like `CrtLib.aws_checksums_crc32(...)`). This `CrtLib` object would ensure `CRT.initRuntime { ... }` is called before invoking the actual C function.